src/shared/seccomp-util.c

```diff
@@ -1051,6 +1051,16 @@ int seccomp_parse_syscall_filter_full(
                                    "Failed to parse system call, ignoring: %s", name);
                         return 0;
                 }
+                if (id == -10189 || id == -10191) {
```

Comment:
I don’t like it either, but for now I like the alternative even less :/ dropping pseudo syscall numbers completely hides those syscalls from `systemctl show`, and I feel like that’s too invasive to do at parse time. (I wouldn’t object if this only happened by the time we apply the filter.)

Reply:
But those numbers look like an implementation details of seccomp. We don't want to bake this into our code.

And I have to say that with that patch things are even more confusing: some syscalls are still hidden from `systemctl show`, but the rules *which* functions are hidden are pretty obscure.

I don't see why we should care about syscalls which are undefined on a given architecture. But if you think it's important to show them, then let's do this properly and adjust the data structures so that it's possible to store those syscalls which don't map to a valid syscall numbers.

Reply:
> But if you think it's important to show them, then let's do this properly and adjust the data structures so that it's possible to store those syscalls which don't map to a valid syscall numbers.

Well – if I understand correctly, that’s exactly the purpose of libseccomp’s pseudo syscall numbers: we can pretend that each syscall corresponds to an integer, and just always pass these integers around, and translate them back to syscall names for display and everything, without having to care which syscalls actually exist and which don’t. It’s just that libseccomp slipped up a bit :(